tests/core/pyspec/eth2spec/test/deneb/sanity/test_blocks.py

```diff
@@ -24,13 +27,23 @@
 )
 
 
-def run_block_with_blobs(spec, state, blob_count, data_gas_used=1, excess_data_gas=1, valid=True):
+def run_block_with_blobs(spec, state, blob_count, tx_count=1, data_gas_used=1, excess_data_gas=1,
+                         non_blob_tx_count=0, rng=random.Random(7777), valid=True):
     yield 'pre', state
 
     block = build_empty_block_for_next_slot(spec, state)
-    opaque_tx, _, blob_kzg_commitments, _ = get_sample_opaque_tx(spec, blob_count=blob_count)
+    txs = []
+    blob_kzg_commitments = []
+    for _ in range(tx_count):
+        opaque_tx, _, commits, _ = get_sample_opaque_tx(spec, blob_count=blob_count)
+        txs.append(opaque_tx)
+        blob_kzg_commitments += commits
+
+    for _ in range(non_blob_tx_count):
+        txs.append(get_random_tx(rng=rng))
```

Comment:
Maybe should use `rng`  to shuffle the `txs` list before applying to the el payload